### PR TITLE
Standby Instrument Improvements - Brightness Adjustment, ATT Reset, Bugs

### PR DESCRIPTION
**Pull request by 2hwk**
## TODOs 
- [x] Brightness Adjustment
  - [x]  Add emissive property to push buttons
  - [x]  Link buttons to screen brightness
  - [x]  Link button brightness to main panel brightness knob
  - [x]  Ungroup DYN_SCREEN textures in model - Worked around for now
  - [x] Better Fonts and more realistic behavior on power off
- [x] ATT RESET
  - [x] 10 second reset timer for Attitude when ATT Reset is pushed
- [x] BUGS
  - [x] Bugs Page for heading and altitude bugs
    - [x] Working Bug selection and setting
  - [x] Altitude Bugs on Altimeter
  - [x] Speed Bugs on IAS

**Summary of Changes**

* Self Test is now 90 seconds to initialise as per documentation
* Event callbacks are now used instead of repeatedly checking on every update (Optimisation!)
* Removed INOP property for all Standby Instrument push buttons (TODO: Bugs/LS)
* Brightness adjustment now available for standby instrument
* Attitude Reset now available for standby instrument (10s ATT Reset)
* Removed placeholder logic, A32NX Utils now used to calculate delta time
* Added TODO logic for when model changes can be implemented
* Added js class templates for bugs/ls for future functionality
* Modified Barometer Knob logic to accommodate bug selection/setting

**Screenshots (if necessary)**

image
image
image
image
image

image
**Known Issues**
* None that can be fixed without model changes, i.e. there is no barometer push knob in the model, nor is there a button hold mode, so these must be improvised. 

**Additional context**
ATT and BARO should not be illuminated, but this is also a bug in the default 320 and is not fixable without model change.

**References**
https://www.a320simulator.be/portfolio/integrated-standby-instrument-system/
https://cdn.discordapp.com/attachments/747724538095075369/761706812327591946/image1.jpg
https://youtu.be/g_Mq0mI50GM
https://www.pprune.org/tech-log/406443-a320-isis-what-adiru-inputs.html
https://aviation.stackexchange.com/questions/57423/what-are-the-bugs-and-rst-buttons-on-the-a320s-isis-for

Discord username (if different from GitHub): 2Cas#1022

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js [hunk 1]
```diff
@@ -1024,16 +1110,16 @@ class A320_Neo_SAI_SelfTestTimer extends HTMLElement {
...
-        this.selfTestSVG.setAttribute("id", "SelfTestSVG");
-        this.selfTestSVG.setAttribute("viewBox", "0 0 600 600");
-        this.selfTestSVG.style.position = "absolute";
-        this.selfTestSVG.style.top = "0%";
-        this.selfTestSVG.style.left = "0%";
-        this.selfTestSVG.style.width = "100%";
-        this.selfTestSVG.style.height = "100%";
-        this.selfTestSVG.style.backgroundColor = "rgba(13,20,35,1)";
-        this.selfTestDiv.appendChild(this.selfTestSVG);
+        const selfTestSVG = document.createElementNS(Avionics.SVG.NS, "svg");
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js:1113]: Use let here, while the actual reference isn't changed, some properties are. Just better use of const and let

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js [hunk 2]
```diff
@@ -1148,8 +1233,188 @@ class A320_Neo_SAI_SelfTestTimer extends HTMLElement {
...
+        if (this.resetTimer >= 0) {
+            this.resetTimer -= dTime / 1000;
+        }
+        if (this.resetTimer <= 0) {
+            this.finishReset();
+        }
+        if (this.resetTimer > 9) {
+            this.st_att_txt.textContent = "ATT " + Math.ceil(this.resetTimer) + "s";
+        }
+        else {
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js:1341]: Don't spread an if else on multiple lines, use `} else {`

Comment (2hwk) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js:1341]: Every if else (in this file at least) is formatted in this way, should they all be changed?

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -25,3 +25,4 @@
 9. [ND] Add DME distances, VOR/ADF needles and functioning ADF2 - @blitzcaster (bltzcstr)
 10. [OVHD] Fixed Battery Indicator Colour - @nathaninnes (Nathan Innes)
 11. [MISC] Removed Fuel Patch from MSFS Update 1.8.3 - @nathaninnes (Nathan Innes)
+12. [MISC] Added standby instrument brightness adjustment and attitude reset - @2hwk (2Cas#1022 on discord)
```
Comment (NathanInnes) [inline .github/CHANGELOG.md:28]: Rebase the changelog and change to correct month

Comment (2hwk) [reply .github/CHANGELOG.md]: Corrected

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js (hunk 2 above)
Comment (AdenFlorian) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js:1341]: you're right 2hwk to match the surrounding code

**Comment by 2hwk** [comment]
I'm putting this back into draft (again!) for now to comply with linting and deal with the merge conflicts. 

I've also been working on implementing the bugs page (working around the missing baro knob functionality has been a challenge) and I think I'd rather have one big PR when it is all completed.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js (hunk 2 above)
Comment (2hwk) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/SAI/A320_Neo_SAI.js]: resolved with linting

**Comment by 2hwk** [comment]
Bugs implemented (heading/speed bugs, not the bad kind)

**Comment by AdenFlorian** [comment]
tested and LGTM! Thanks!

**Pull request merged**